packages/select/src/components/multi-select/multiSelect.tsx

```diff
@@ -294,33 +325,41 @@ export class MultiSelect<T> extends AbstractPureComponent<MultiSelectProps<T>, M
                     // Normally, Popover would also need to attach its own `onKeyDown` handler via `targetProps`,
                     // but in our case we fully manage that interaction and listen for key events to open/close
                     // the popover, so we elide it from the DOM.
-                    onKeyDown: this.getTagInputKeyDownHandler(handleKeyDown),
-                    onKeyUp: this.getTagInputKeyUpHandler(handleKeyUp),
+                    onKeyDown:
+                        this.props.customTarget != null ? undefined : this.getTagInputKeyDownHandler(handleKeyDown),
+                    onKeyUp: this.props.customTarget != null ? undefined : this.getTagInputKeyUpHandler(handleKeyUp),
                     ref,
                     role: "combobox",
                 },
-                <TagInput
-                    placeholder={placeholder}
-                    rightElement={maybeClearButton}
-                    {...tagInputProps}
-                    className={classNames(Classes.MULTISELECT, tagInputProps.className)}
-                    disabled={disabled}
-                    fill={fill}
-                    inputRef={this.refHandlers.input}
-                    inputProps={inputProps}
-                    inputValue={listProps.query}
-                    onAdd={this.getTagInputAddHandler(listProps)}
-                    onInputChange={listProps.handleQueryChange}
-                    onRemove={this.handleTagRemove}
-                    values={selectedItems.map(this.props.tagRenderer)}
-                />,
+                this.props.customTarget != null ? (
+                    this.props.customTarget
+                ) : (
+                    <TagInput
+                        placeholder={placeholder}
+                        rightElement={maybeClearButton}
+                        {...tagInputProps}
+                        className={classNames(Classes.MULTISELECT, tagInputProps.className)}
+                        disabled={disabled}
+                        fill={fill}
+                        inputRef={this.refHandlers.input}
+                        inputProps={inputProps}
+                        inputValue={listProps.query}
+                        onAdd={this.getTagInputAddHandler(listProps)}
+                        onInputChange={listProps.handleQueryChange}
+                        onRemove={this.handleTagRemove}
+                        values={selectedItems.map(this.props.tagRenderer)}
+                    />
+                ),
             );
         };
 
     private handleItemSelect = (item: T, evt?: React.SyntheticEvent<HTMLElement>) => {
         if (this.input != null) {
             this.input.focus();
         }
+        if (this.popoverInputForCustomTarget != null) {
+            this.popoverInputForCustomTarget.focus();
```

Comment:
is this really needed? I wonder if this similar code above on the `input` is what causes the behavior we often hear complaints about where the list scrolls back to the top after selecting an item

Reply:
I think this is a useful feature fwiw, and keeps the same behavior as TagInput. Feels kind of snappy just being able to search and use arrows + enter to select items, without having to use your mouse to refocus on the input after every selection

the complaint your describing is unrelated to this, that's likely due to after every selection the active item returning to index 0 in the list and being scrolled into view due to `scrollToActiveItem` prop being default true.